airflow/api_connexion/schemas/enum_schemas.py

```diff
@@ -0,0 +1,24 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from marshmallow import fields, validate
+from marshmallow.schema import Schema
+
+
+class DagState(Schema):
+    """DagState schemagit"""
+    state = fields.Str(validate=validate.OneOf(["success", "running", "failed"]))
```

Comment:
I used the idea on this link below to work on this but I am unsure if it is the right way to do it.
https://github.com/marshmallow-code/marshmallow/issues/1470